Shared/Notifications/NotificationsManager.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  NotificationManager.swift
+//  Eatery
+//
+//  Created by Ethan Fine on 2/17/20.
+//  Copyright © 2020 CUAppDev. All rights reserved.
+//
+
+import SwiftyUserDefaults
+import UserNotifications
+
+class NotificationsManager {
+    
+    static let shared = NotificationsManager()
+    
+    private let favoriteMenuItemNotifTitle = "Grub on one of your favorite foods!"
+    private let notifContentDateKey = "date"
+    private let notifContentEateriesKey = "eateryNames"
+    private let notifContentEventNameKey = "eventName"
+    private let notifContentMenuItemKey = "menuItemName"
+    /// Maps menu item names to an an array of scheduled notification requests
+    private var scheduledNotifications = [String : [UNNotificationRequest]]()
+    
+    /// Requests authorization to the user for notifications
+    func requestAuthorization() {
+        var authorizationOptions: UNAuthorizationOptions = [.alert]
+        if #available(iOS 13.0, *) {
+            authorizationOptions.insert(.announcement)
+        }
+        UNUserNotificationCenter.current().requestAuthorization(options: authorizationOptions) { (granted, error) in }
```

Comment:
might be worth making a `private let notificationCenter = UNUserNotificationCenter.current()` as a class field so you don't have to keep on repeating it